Chap_API_Server.tex

```diff
@@ -45,9 +45,11 @@ \subsection{\code{PMIx_generate_regex}}
 
 Given a comma-separated list of \refarg{input} values, generate a reduced size representation of the input that can be passed down to the \ac{PMIx} server library's \refapi{PMIx_server_register_nspace} \ac{API} for parsing. The order of the individual values in the \refarg{input} string is preserved across the operation. The caller is responsible for releasing the returned data.
 
-\adviceuserstart
-The returned representation may be an arbitrary array of bytes as opposed to a valid NULL-terminated string. However, the method used to generate the representation shall be identified with a colon-delimited string at the beginning of the output. For example, an output starting with \code{"pmix:"} indicates that the representation is a \ac{PMIx}-defined regular expression. In contrast, an output starting with \code{"blob:"} is a compressed binary array.
-\adviceuserend
+\advicermstart
+The returned representation may be an arbitrary array of bytes as opposed to a valid NULL-terminated string. However, the method used to generate the representation shall be identified with a colon-delimited string at the beginning of the output. For example, an output starting with \code{"pmix:"} indicates that the representation is a \ac{PMIx}-defined regular expression represented as a NULL-terminated string. In contrast, an output starting with \code{"blob:\textbackslash0size=1234:"} is a compressed binary array.
```

Comment:
That gets rather implementation dependent, doesn't it? I was just giving an example of how someone might encode a regex, but that is just how I chose to do it - there is nothing to say that someone else will do it the same way.

You don't need to explicitly state the size in the "pack/unpack" routines because I created a new `PMIX_REGEX` data type. It is up to the implementation to identify whatever size information they encoded into the array and "do the right thing". The user has no idea of the size nor any need to know it.

Reply:
Ah yes. That's want I wanted to confirm. That they didn't need to know the size of the buffer for packing/unpacking. Since they are packing `1` regex then that's all they need to know. The PRI figures out the side from however it encoded it internally.

Ok that works for me. Thanks for walking me through it.

Reply:
To avoid confusion about the include of the size what do you think about leaving it out? just having it listed as `\code{"blob:"}`

Reply:
Done - plus added some more verbage to help.